lib/mayaUsd/ufe/PullPushCommands.cpp

```diff
@@ -0,0 +1,371 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "PullPushCommands.h"
+
+#include "Utils.h"
+
+#include <mayaUsd/fileio/primUpdaterManager.h>
+#include <mayaUsd/undo/OpUndoInfoRecorder.h>
+#include <mayaUsd/undo/UsdUndoManager.h>
+#include <mayaUsd/utils/util.h>
+
+#include <maya/MArgParser.h>
+#include <maya/MGlobal.h>
+#include <maya/MStringArray.h>
+#include <maya/MSyntax.h>
+#include <ufe/path.h>
+#include <ufe/pathString.h>
+
+#include <algorithm>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+namespace {
+
+// Reports an error to the Maya scripting console.
+void reportError(const MString& errorString) { MGlobal::displayError(errorString); }
+
+// Reports an error based on the status. Reports nothing on success.
+MStatus reportError(MStatus status)
+{
+    switch (status.statusCode()) {
+    case MStatus::kNotFound: reportError("No object were provided."); break;
+    case MStatus::kInvalidParameter: reportError("Invalid object path."); break;
+    case MStatus::kUnknownParameter: reportError("Invalid parameter."); break;
+    case MStatus::kSuccess: break;
+    default: reportError("Command parsing error."); break;
+    }
+
+    return status;
+}
+
+// Parses a string as a UFE path, detecting invalid paths and empty paths.
+MStatus parseArgAsUfePath(const MString& arg, Ufe::Path& outputPath)
```

Comment:
Do you really need to tell the difference between MS::kNotFound and MS::kInvalidParameter?  May as well return an empty Ufe::Path for the error case if you don't care.